### ISSUE TITLE
Warning in IntelliJ EAP: `ActionUpdateThread.OLD_EDT` is deprecated and going to be removed soon.

### ISSUE THREAD

**Issue by jiridanek** [wallabyjs/public#3327]
This is just a deprecation warning, no big deal.

```
com.intellij.diagnostic.PluginException: `ActionUpdateThread.OLD_EDT` is deprecated and going to be removed soon. 'com.wallaby.javascript.actions.StartExclusiveTestRunAction' must override `getActionUpdateThread` and chose EDT or BGT. See ActionUpdateThread javadoc. [Plugin: wallaby.js]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:90)
	at com.intellij.diagnostic.PluginException.reportDeprecatedUsage(PluginException.java:125)
	at com.intellij.openapi.actionSystem.ActionUpdateThreadAware.getActionUpdateThread(ActionUpdateThreadAware.java:21)
	at com.intellij.openapi.actionSystem.AnAction.getActionUpdateThread(AnAction.java:198)
```

```
IntelliJ IDEA 2024.1 EAP (Ultimate Edition)
Build #IU-241.9959.31, built on January 26, 2024
Licensed to IntelliJ IDEA EAP user: @jiridanek
Expiration date: February 25, 2024
Runtime version: 21.0.2+13-b341.4 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 6.6.13-200.fc39.x86_64
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 2048M
Cores: 12
Registry:
  debugger.new.tool.window.layout=true
  freeze.reporter.profiling=false
  ide.experimental.ui=true
Non-Bundled Plugins:
  com.jetbrains.plugins.ini4idea (241.9959.38)
  org.antlr.intellij.plugin (1.22)
  org.intellij.plugins.hcl (241.9959.31)
  com.jetbrains.rust (241.23135.127)
  wallaby.js (1.0.284)
  com.jetbrains.space (241.9959.38)
  quokka.js (1.0.430)
  org.jetbrains.plugins.go-template (241.9959.31)
  com.intellij.ideolog (222.3.1.0)
  name.kropp.intellij.makefile (241.9959.31)
  ir.msdehghan.plugins.ansible (0.92)
  org.jetbrains.plugins.go (241.9959.31)
  AceJump (3.8.18)
  robocorp.lsp.intellij (1.11.0)
  Pythonid (241.9959.31)
  com.intellij.grazie.pro (0.3.276)
  com.intellij.plugins.kdekeymap (213.4250.391)
  com.intellij.nativeDebug (241.9959.31)
  intellij.haskell (1.0.0-beta88)
  com.intellij.plugin.adernov.powershell (2.3.1)
  org.asciidoctor.intellij.asciidoc (0.41.6)
  com.intellij.bigdatatools.core (241.9959.31)
  com.intellij.bigdatatools.binary.files (241.9959.31)
  com.intellij.bigdatatools.rfs (241.9959.31)
  com.intellij.bigdatatools.metastore.core (241.9959.31)
  dev.blachut.svelte.lang (241.9959.31)
  mobi.hsz.idea.gitignore (4.5.2)
  com.carbonblack.intellij.rpmspec (2.2.0)
  com.intellij.bigdatatools.spark (241.9959.31)
Kotlin: 241.9959.31-IJ
Current Desktop: KDE
```

**Comment by smcenlly** [comment]
Thanks for reporting the issue. We'll investigate and update in the next few weeks.

**Comment by smcenlly** [comment]
We've investigated fixing the deprecation warning. In order to fix the issue, we would need to update our usage of the JetBrains API which is was released with 2022.3+ (30 November 2022).

We aim to support **at least** the last 2 years of editors released by JetBrains (currently we support 2.5 years). Fixing this issue would decrease the editors we support to less than 1.2 years.

We appreciate you raising the issue and we will fix the issue at a later time. Marking this issue on hold for now.

**Comment by Alexsey** [comment]
After updating the WebStorm to 2024.1 I'm having tons of these warnings, which is quite annoying. Considering that you would need to migrate anyway, maybe it's possible to use the new API for the 2024.1+ IDE?

![Screenshot 2024-04-09 at 19 04 39](https://github.com/wallabyjs/public/assets/6392013/9880a7b4-618c-4641-a0cf-dcaf3ee35c7f)

**Comment by mburnell** [comment]
@Alexsey Thanks for the update. While we'd still like to maintain the compatibility mentioned in our earlier response, we understand this has become more annoying, and will see what we can do to fix or suppress this warning in newer versions.

**Comment by mburnell** [comment]
This issue has been addressed in version `1.0.286`; after updating to this version of the Wallaby plugin, this warning/error should no longer occur.
